### PR TITLE
[vLLM] Fix cpu sampling

### LINKED ISSUE #4331: [vLLM plugin][Bug] CPU samling failure
vLLM plugin create sampling metadata on `cpu` when `cpu_sampling` is enabled. However, model runner is always using same `sample_from_logits` function. This cause failure as the sampling data is created on cpu whereas logits are still on device. 
`sample_from_logits_cpu` moves data to cpu but keep the output on cpu which may cause failure later.

### PR DESCRIPTION
**Pull request by mmanzoorTT**
### Ticket
closes #4331 

### Problem description
model runner always process sampling on device even if cpu_sampling is enabled. 

### What's changed
Update the function call to use appropriate functionality.

### CODE REVIEW

File: integrations/vllm_plugin/vllm_tt/model_runner.py [hunk 1]
```diff
@@ -1380,7 +1380,9 @@ def sample_tokens(
                     require_struct_decoding, grammar_bitmask_padded, logits, arange
                 )
 
-            selected_token_ids = self.sample_from_logits(logits, tpu_sampling_metadata)
+            selected_token_ids = self.sample_from_logits_func(
+                logits, tpu_sampling_metadata
```
Comment (AleksKnezevic) [inline integrations/vllm_plugin/vllm_tt/model_runner.py]: Doesn't need to be in this PR, but we should rename tpu_sampling_metadata

**Review by AleksKnezevic — APPROVED** [review]
